payload/Library/Scripts/Install or Defer.sh

```diff
@@ -421,6 +428,9 @@ else
 fi
 echo "Maximum deferral time: $(convert_seconds "$MAX_DEFERRAL_TIME")"
 
+# Check for updates, exit if none found, otherwise cache locally and continue.
+check_for_updates
```

Comment:
This change addresses an extreme edge case: I found that my update test Macs, which have a custom deferral period of 1 hour, were running down the clock by defining the start time before starting the cache download. macOS updates are only getting bigger, and as more folks are currently working away from corporate offices, they will potentially be running this script on slower home networks and/or VPN, so I think it makes sense to not start that countdown until the update is cached for the first time.